core/include/lifo.h

```diff
@@ -13,6 +13,7 @@
  * @file
  * @brief   LIFO buffer API, read long description carefully
  * @author  Heiko Will <hwill@inf.fu-berlin.de>
+ * @deprecated this will be removed with release 2021.07
```

Comment:
Please provide the reasoning for removal. See the [deprecated list](https://doc.riot-os.org/deprecated.html) for some examples.